modules/ding_event/ding_event.admin.inc

```diff
@@ -50,12 +70,16 @@ function ding_event_admin_currency_submit($form, &$form_state) {
 }
 
 /**
- * Implement hook_permission
+ * Validate input and update field_instance
+ *
+ * @param array $form
+ *   An associative array containing the structure of the form.
+ * @param array $form_state
+ *   A keyed array containing the current state of the form.
  */
-function ding_event_permission() {
-  return array(
-    'administer event settings' => array(
-      'title' => t('Administer event settings')
-    ),
-  );
+function ding_event_admin_validate(array $form, array &$form_state) {
+  if (!ctype_digit($form_state['values']['ding_event_frontend_items_per_page'])) {
```

Comment:
It seems like this could be replaced with setting [`#element_validate = array('element_validate_integer_positive')`](https://api.drupal.org/api/drupal/includes%21form.inc/function/element_validate_integer_positive/7) on the element.